### PR TITLE
feat(taiko-client): support `TaikoL1.proposeBlocksV2`

### LINKED ISSUE #18113: feat(taiko-client): support `TaikoL1.proposeBlocks`

### PR DESCRIPTION
**Pull request by davidtaikocha**

### CODE REVIEW

**Review by YoGhurt111 — APPROVED** [review]

File: packages/taiko-client/proposer/proposer.go [hunk 1]
```diff
@@ -291,46 +293,60 @@ func (p *Proposer) ProposeOp(ctx context.Context) error {
...
+	if err != nil {
+		return err
+	}
 
-		g.Go(func() error {
-			txListBytes, err := rlp.EncodeToBytes(txs)
+	// If the current L2 chain is before ontake fork, propose the transactions lists one by one.
+	if !p.chainConfig.IsOntake(new(big.Int).SetUint64(state.B.NumBlocks)) {
+		g, gCtx := errgroup.WithContext(ctx)
+		for _, txs := range txLists[:utils.Min(p.MaxProposedTxListsPerEpoch, uint64(len(txLists)))] {
```
Comment (mask-pp) [inline packages/taiko-client/proposer/proposer.go:311]:
```suggestion
		for _, txs := range txLists[:utils.Min(p.MaxProposedTxListsPerEpoch, uint64(len(txLists)))] {
		    txs := txs
```

File: packages/taiko-client/proposer/proposer.go [hunk 2]
```diff
@@ -372,10 +388,76 @@ func (p *Proposer) ProposeTxList(
...
+
+// ProposeTxListOntake proposes the given transactions lists to TaikoL1 smart contract.
+func (p *Proposer) ProposeTxListOntake(
+	ctx context.Context,
+	txLists []types.Transactions,
+) error {
+	var (
+		proverAddress     = p.proposerAddress
+		txListsBytesArray [][]byte
+		txNums            []uint8
```
Comment (mask-pp) [inline packages/taiko-client/proposer/proposer.go:407]:
```suggestion
		txNums            []int
```

File: packages/taiko-client/proposer/proposer.go [hunk 3]
```diff
@@ -372,10 +388,76 @@ func (p *Proposer) ProposeTxList(
...
+		totalTxs          uint8
```
Comment (mask-pp) [inline packages/taiko-client/proposer/proposer.go:408]:
```suggestion
		totalTxs          int
```

File: packages/taiko-client/pkg/rpc/methods.go [hunk 4]
```diff
@@ -383,7 +383,7 @@ func (c *Client) GetPoolContent(
 		ctx,
 		l2Head,
 		l1Head.Number,
-		chainConfig.IsOntake(l2Head.Number),
+		chainConfig.IsOntake(new(big.Int).Add(l2Head.Number, common.Big1)),
```
Comment (mask-pp) [inline packages/taiko-client/pkg/rpc/methods.go:386]: Why plus one?

File: packages/taiko-client/proposer/proposer.go (hunk 1 above)
Comment (davidtaikocha) [reply packages/taiko-client/proposer/proposer.go:311]: Why we need to do that here?

File: packages/taiko-client/proposer/proposer.go (hunk 2 above)
Comment (davidtaikocha) [reply packages/taiko-client/proposer/proposer.go:407]: What's the difference?

**Review by mask-pp — APPROVED** [review]

**Pull request merged**